File: app/src/main/java/net/ducksmanager/util/DraggableRelativeLayout.java
package net.ducksmanager.util;

import android.view.MotionEvent;
import android.view.View;
import android.widget.RelativeLayout;

public class DraggableRelativeLayout {
    public static void makeDraggable(RelativeLayout view) {
        view.setOnTouchListener(new View.OnTouchListener() {
            float dX, dY;
            @Override
            public boolean onTouch(View view, MotionEvent event) {
                switch (event.getAction()) {
                    case MotionEvent.ACTION_DOWN:
                        dX = view.getX() - event.getRawX();
                        dY = view.getY() - event.getRawY();
                        break;

                    case MotionEvent.ACTION_MOVE:
                        view.animate()
                            .x(event.getRawX() + dX)
                            .y(event.getRawY() + dY)
                            .setDuration(0)
                            .start();
                        break;
                    default:
                        return false;
                }
                return true;
            }
        });
    }
}

File: app/src/main/java/net/ducksmanager/util/DraggableRelativeLayout.kt
package net.ducksmanager.util

import android.annotation.SuppressLint
import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener

object DraggableRelativeLayout {
    @SuppressLint("ClickableViewAccessibility")
    fun makeDraggable(view: View) {
        view.setOnTouchListener(object : OnTouchListener {
            var dX = 0f
            var dY = 0f
            override fun onTouch(view: View, event: MotionEvent): Boolean {
                when (event.action) {
                    MotionEvent.ACTION_DOWN -> {
                        dX = view.x - event.rawX
                        dY = view.y - event.rawY
                    }
                    MotionEvent.ACTION_MOVE -> view.animate()
                        .x(event.rawX + dX)
                        .y(event.rawY + dY)
                        .setDuration(0)
                        .start()
                    else -> return false
                }
                return true
            }
        })
    }
}